### PR TITLE
Remove the costly and unnecessary setSlobrokId interface.

### PR DESCRIPTION
**Pull request by baldersheim**
@havardpe PR

### CODE REVIEW

**Comment by baldersheim** [comment]
Merge to unbreak factory

File: vespaclient/src/vespa/vespaclient/vesparoute/application.cpp [hunk 1]
```diff
@@ -45,33 +45,29 @@ Application::Main()
...
+        if ( ! table) {
             throw config::InvalidConfigException(vespalib::make_string("There is no routing table for protocol '%s'.",
                                                                        _params.getProtocol().c_str()));
         }
-        for (std::vector<std::string>::iterator it = _params.getHops().begin();
-             it != _params.getHops().end(); ++it)
-        {
-            if (table->getHop(*it) == NULL) {
+        for (const std::string & hop : _params.getHops()) {
+            if (table->getHop(hop) == NULL) {
```
Comment (havardpe) [inline vespaclient/src/vespa/vespaclient/vesparoute/application.cpp]: nullptr

File: messagebus/src/vespa/messagebus/network/rpcnetworkparams.h [hunk 2]
```diff
@@ -69,15 +70,8 @@ class RPCNetworkParams {
     }
 
     /**
-     * Sets of the slobrok config.
      *
-     * @param slobrokConfigId The new config.
-     * @return This, to allow chaining.
      */
-    RPCNetworkParams &setSlobrokConfig(const config::ConfigUri & slobrokConfig) {
```
Comment (havardpe) [inline messagebus/src/vespa/messagebus/network/rpcnetworkparams.h]: why was this so expensive?

File: vespaclient/src/vespa/vespaclient/vesparoute/params.h [hunk 3]
```diff
@@ -92,13 +92,17 @@ class Params {
...
+    void setSlobrokId(const std::string & id) { _slobrokConfigId = id; }
+
 private:
     mbus::RPCNetworkParams   _rpcParams;
     std::vector<std::string> _hops;
     std::vector<std::string> _routes;
     std::string              _documentTypesConfigId;
     std::string              _routingConfigId;
     std::string              _protocol;
+    std::string              _slobrokConfigId;
```
Comment (havardpe) [inline vespaclient/src/vespa/vespaclient/vesparoute/params.h]: how are you able to get this into the params constructor?

**Review by havardpe — COMMENTED** [review]
there are too many changes for me to see/understand what has changed (setting the config uri in the constructor instead of a setter does not seem to change very much, neither does the smart pointering)

**Comment by havardpe** [comment]
auch. all of the above

**Pull request merged**